src/mono/wasm/Makefile

```
@@ -0,0 +1,104 @@
+TOP=$(realpath $(CURDIR)/../../..)
+-include Make.config
+
+#
+# These variables are set by wasm.targets
+#
+EMSDK_PATH?=emsdk
+CONFIG?=Release
+BINDIR?=$(TOP)/artifacts/bin
+OBJDIR?=$(TOP)/artifacts/obj
+PINVOKE_TABLE?=$(TOP)/artifacts/obj/mono/Browser.wasm.$(CONFIG)/wasm/pinvoke-table.h
+MONO_BIN_DIR?=$(BINDIR)/mono/Browser.wasm.$(CONFIG)
+
+all: build-native
+
+#
+# EMSCRIPTEN SETUP
+#
+#  If EMSDK_PATH is not set by the caller, download and setup a local emsdk install.
+#
+
+EMSCRIPTEN_VERSION=1.39.11
+EMSDK_LOCAL_PATH=emsdk
+EMCC=source $(CURDIR)/emsdk_env.sh && emcc
+
+emsdk:
+	git clone https://github.com/emscripten-core/emsdk $(EMSDK_PATH)
+
+$(EMSDK_PATH)/.emscripten: | $(EMSDK_PATH)
+	touch $@
+
+.stamp-wasm-install-and-select-$(EMSCRIPTEN_VERSION): .stamp-wasm-checkout-and-update-emsdk $(EMSDK_PATH)/.emscripten
+	cd emsdk && ./emsdk install $(EMSCRIPTEN_VERSION)
+	cd emsdk && ./emsdk activate --embedded $(EMSCRIPTEN_VERSION)
+	touch $@
+
+.stamp-wasm-checkout-and-update-emsdk: | $(EMSDK_PATH)
+	cd emsdk && git reset --hard && git clean -xdff && git pull
+	touch $@
+
+ifeq ($(EMSDK_PATH),emsdk)
+provision-wasm: .stamp-wasm-install-and-select-$(EMSCRIPTEN_VERSION)
+else
+provision-wasm:
+endif
+
+# emsdk_env.sh calls emsdk construct_env which is a bit slow so make a copy
+emsdk_env.sh: | provision-wasm
+	cd $(EMSDK_PATH) && ./emsdk construct_env $(CURDIR)/emsdk_env.sh
+
+MONO_OBJ_DIR=$(OBJDIR)/mono/Browser.wasm.$(CONFIG)
+SYS_NATIVE_DIR=$(BINDIR)/native/netcoreapp5.0-Browser-$(CONFIG)-wasm/
```

@vargaz this should fix the current build failure:

```suggestion
SYS_NATIVE_DIR=$(BINDIR)/native/net5.0-Browser-$(CONFIG)-wasm/
```

Ideally we'd not deal with this in Makefile...